tools/cloud_run/event_driven_bq_ingest/src/main/java/com/example/cloudrun/JobAccessor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloudrun;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.ExternalTableDefinition;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.JobId;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.JobConfiguration;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+@Log4j2
+public class JobAccessor {
+
+  public static List<String> checkJobCompeletion(PubSubMessageData pubSubMessageData) {
+    log.info("Resource Name:{}", pubSubMessageData.getProtoPayload().getResourceName());
+
+    try {
+      String resourceName = pubSubMessageData.getProtoPayload().getResourceName();
+      String[] parsedName = resourceName.split("/");
+      String jobName = parsedName[parsedName.length - 1];
+
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+      JobId jobId = JobId.of(jobName);
+      Job job = bigquery.getJob(jobId);
+
+      List<String> allSourceUris = new ArrayList<String>();
+      // check for error
+      if (job.isDone()) {
```

Comment:
This does not mean the job was successful, only that that job completed. 
You must check the status of the job as follows:

if(job.isDone() && job.getStatus().getError() == null)

Source: https://googleapis.dev/java/google-cloud-bigquery/latest/com/google/cloud/bigquery/BigQueryError.html